middleware/src/util.go

```diff
@@ -164,3 +167,66 @@ func determineEnableValue(enable rpcmessages.ToggleSetting) string {
 	}
 	return "disable"
 }
+
+// getStringFromRedis returns a string for a given Redis key.
+// If an error occours a ErrorResponse is returned with a ErrorRedisError as error code.
+func (middleware *Middleware) getStringFromRedis(key redis.BaseRedisKey) (value string, errResp rpcmessages.ErrorResponse) {
+	log.Printf("getting '%s' as string from Redis\n", key)
+	value, err := middleware.redisClient.GetString(key)
+	if err != nil {
+		return "", rpcmessages.ErrorResponse{
+			Success: false,
+			Message: fmt.Errorf("could not get %s from redis: %s", key, err.Error()).Error(),
+			Code:    rpcmessages.ErrorRedisError,
+		}
+	}
+	return value, rpcmessages.ErrorResponse{Success: true}
```

Comment:
if ErrorResponse was `nil` in case of success, this line would also look more natural.